src/edge/edgeManager.ts

```diff
@@ -319,6 +372,101 @@ export class EdgeManager {
         return { repositoryName, imageName };
     }
 
+    private async updateRegistrySettings(address: string, registries: any, envFile: string): Promise<{registries: string, usernameEnv: string, passwordEnv: string}> {
+        let usernameEnv;
+        let passwordEnv;
+        const lowerCase = address.toLowerCase();
+        if (lowerCase === "localhost" || lowerCase.startsWith("localhost:")) {
+            return {registries, usernameEnv, passwordEnv};
+        }
+        await Utility.loadEnv(envFile);
+        const {exists, keySet} = this.checkAddressExist(address, registries);
+
+        if (!exists) {
+            const addressKey = Utility.getAddressKey(address, keySet);
+            usernameEnv = `CONTAINER_REGISTRY_USERNAME_${addressKey}`;
+            passwordEnv = `CONTAINER_REGISTRY_PASSWORD_${addressKey}`;
+            const newRegistry = `{
+                "username": "$${usernameEnv}",
+                "password": "$${passwordEnv}",
+                "address": "${address}"
+            }`;
+            if (!registries) {
+                registries = {};
+            }
+            registries[addressKey] = JSON.parse(newRegistry);
+        }
+        return {registries, usernameEnv, passwordEnv};
+    }
+
+    private async writeRegistryCredEnv(address: string, envFile: string, usernameEnv: string, passwordEnv: string): Promise<void> {
+        if (!usernameEnv) {
+            return;
+        }
+
+        if (address.endsWith(".azurecr.io")) {
+            await this.populateACRCredential(address, envFile, usernameEnv, passwordEnv);
+        } else {
+            await this.populateStaticEnv(envFile, usernameEnv, passwordEnv);

```

Comment:
If it is a public docker hub registry, will the username and password still be added to .env file?

Reply:
Yes for now. I struggling the workflow yesterday, And found this can save a message prompt. Besides, if we don't do so. User may mostly not add the credential, and hard to detect